### ISSUE TITLE
Bug: #15961 was incorrectly closed. There is an error in eslint for node.js public class variables

### ISSUE THREAD

**Issue by TotallyInformation** [eslint/eslint#15964]
### Environment

Node version: v12
npm version: N/A
Local ESLint version: 8.16.0
Global ESLint version: 8.16.0
Operating System: Windows 10

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

See https://github.com/eslint/eslint/issues/15961
</details>

```
class UibPackages {
    // TODO: Replace _XXX with #XXX once node.js v14 is the minimum supported version
    /** Flag to indicate whether setup() has been run (ignore the false eslint error)
     * @type {boolean}
     * @protected 
     */
    _isConfigured = false

    /** @type {Array<string>} Updated by updateMergedPackageList which is called first in setup and then in various updates */
    mergedPkgMasterList = []
    /** @type {string} The name of the package.json file 'package.json' */
    packageJson = 'package.json'
    /** @type {*} The uibRoot package.json contents */
    uibPackageJson
    /** @type {string} Get npm's global install location */
    globalPrefix // set in constructor

    constructor() {
      this.fred = 'hi there'
      console.log(this.packageJson)
    }
    
    showMe() {
      console.log(this.packageJson, this.fred)
    }
}

const x = new UibPackages()

console.log("Hello, World!", x.showMe(), x.packageJson);
```

### What did you expect to happen?

In node.js, public class variables were supported in v12+, eslint does not allow them until ECMA 13 which is incorrect. They should be supported from at least ECMA 10 if not 9 for node.js.

Please don't blindly close this as with #15961, I have tested this and it works fine in node.js v12.14 as shown by https://onecompiler.com/nodejs/3y5zhpz3d

### What actually happened?

Please see the incorrectly closed #15961 which contains the details of the error that is shown.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

https://github.com/eslint/eslint/issues/15961

**Comment by mdjermanovic** [comment]
`parserOptions.ecmaVersion` is a version of the ECMAScript Language Specification.

Class fields are added in ECMAScript 2022 Language Specification: https://tc39.es/ecma262/2022/

They didn't exist in ECMAScript 2021 Language Specification: https://tc39.es/ecma262/2021/

You can check which feature was added in which version (the "Expected Publication Year" column) here:

https://github.com/tc39/proposals/blob/main/finished-proposals.md

Some environments may support features that have not yet been added to the language specification. If your target environments support ECMAScript 2022, you can set `parserOptions: { ecmaVersion: 2022 }`.
